src/robot/reporting/xunitwriter.py

```diff
@@ -70,7 +63,7 @@ def visit_test(self, test):
         self._writer.start('testcase',
                            {'classname': test.parent.longname,
                             'name': test.name,
-                            'time': self._time_as_seconds(test.elapsedtime)})
+                            'time': f'{test.elapsedtime / 1000 :.3f}'})
```

Comment:
In my opinion `_time_as_seconds` was a good helper method because it made it clear what the code does. `f'{test.elapsedtime / 1000 :.3f}'` is likely not as obvious.

Now that I look at this again, I got a feeling that `f'{millis / 1000:.3f}'`is kind-of abusing f-strings. There's no template, it's only used for formatting and using `format(millis / 1000, '.3f')` would be better. Alternatively could also use `str(round(millis / 1000, 3))`.

Reply:
I was also thinking that removing `_time_as_seconds` helper might be too much. I'll play around with `format(millis / 1000, '.3f')` and `str(round(millis / 1000, 3))` when getting back the helper.